packages/components/src/components/data-list/data-list.jsx

```diff
@@ -1,74 +1,81 @@
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import React from 'react';
-import { NavLink } from 'react-router-dom';
 import { TransitionGroup } from 'react-transition-group';
 import { CellMeasurer, CellMeasurerCache } from 'react-virtualized/dist/es/CellMeasurer';
 import { AutoSizer } from 'react-virtualized/dist/es/AutoSizer';
 import { List } from 'react-virtualized/dist/es/List';
 import { isMobile, isDesktop } from '@deriv/shared';
 import DataListCell from './data-list-cell.jsx';
+import DataListRowWrapper from './data-list-row-wrapper.jsx';
 import ThemedScrollbars from '../themed-scrollbars';
 
-class DataList extends React.PureComponent {
-    state = {
-        scrollTop: 0,
-        isScrolling: false,
-    };
-    items_transition_map = {};
+const DataList = React.memo(props => {
+    const [scrollTop, setScrollTop] = React.useState(0);
+    const [isScrolling, setIsScrolling] = React.useState(false);
+    const [loading, setLoading] = React.useState(true);
+
+    const cache = React.useRef();
+    const list_ref = React.useRef();
+    const items_transition_map = {};
+    const is_dynamic_height = !props.getRowSize;
 
-    constructor(props) {
-        super(props);
-        const { getRowSize } = props;
-        this.is_dynamic_height = !getRowSize;
+    const trackItemsForTransition = () => {
+        props.data_source.forEach((item, index) => {
+            const row_key = props.keyMapper?.(item) || `${index}-0`;
+            items_transition_map[row_key] = true;
+        });
+    };
 
-        if (this.is_dynamic_height) {
-            this.cache = new CellMeasurerCache({
+    React.useEffect(() => {
+        if (is_dynamic_height) {
+            cache.current = new CellMeasurerCache({
                 fixedWidth: true,
-                keyMapper: row_index => this.props.keyMapper?.(this.props.data_source[row_index]) || row_index,
+                keyMapper: row_index => {
+                    if (row_index < props.data_source.length)
+                        return props.keyMapper?.(props.data_source[row_index]) || row_index;
+                    return row_index;
+                },
             });
         }
-        this.trackItemsForTransition();
-    }
+        trackItemsForTransition();
+        setLoading(false);
+    }, []);
```

Comment:
Did you check output of `npm run test` for `eslint` warnings?

Reply:
I want this `useEffect` to run just one time at the first render so the second param should be `[]`.